docs/source/api/data/batch.rst

```diff
@@ -0,0 +1,12 @@
+.. _batch_objects:
+
+Batch Objects
+=============
+
+.. currentmodule:: mlx_graphs.data.batch
+
+
+.. autosummary::
+	:toctree: _autosummary
+
+	GraphDataBatch
```

Comment:
@francescofarina do you know how to consider `__incr__` and `__cat_dim__` methods in the docs?

Reply:
We'll likely have to write a custom template (right now I wrote only this docs/source/_templates/custom-class-template.rst for the nn module to include the `__call__`). Why do you want to include them in the doc?

Reply:
It's important to have these in the docs as the user can override them for custom batching.

Reply:
Ok, then a custom template is the way to go.

Reply:
ok we'll handle that in another PR